nautobot/extras/datasources/git.py

```diff
@@ -822,27 +823,67 @@ def delete_git_config_context_schemas(repository_record, job_result, preserve=()
 #
 
 
+def refresh_code_from_repository(repository_slug, consumer=None, skip_reimport=False):
+    """
+    After cloning/updating a GitRepository on disk, call this function to reload and reregister the repo's Python code.
+
+    Args:
+        repository_slug (str): Repository directory in GIT_ROOT that was refreshed.
+        consumer (celery.worker.Consumer): Celery Consumer to update as well
+        skip_reimport (bool): If True, unload existing code from this repository but do not re-import it.
+    """
+    if settings.GIT_ROOT not in sys.path:
+        sys.path.append(settings.GIT_ROOT)
+
+    app = consumer.app if consumer is not None else celery_app
+    # TODO: This is ugly, but when app.use_fast_trace_task is set (true by default), Celery calls
+    # celery.app.trace.fast_trace_task(...) which assumes that all tasks are cached and have a valid `__trace__()`
+    # function defined. In theory consumer.update_strategies() (below) should ensure this, but it doesn't
+    # go far enough (possibly a discrepancy between the main worker process and the prefork executors?)
+    # as we can and do still encounter errors where `task.__trace__` is unexpectedly None.
+    # For now, simply disabling use_fast_trace_task forces the task trace function to be rebuilt each time,
+    # which avoids the issue at the cost of very slight overhead.
+    app.use_fast_trace_task = False
```

Comment:
Can this be set as a class attribute on `nautobot.core.celery.NautobotCelery`?

Reply:
Not as far as I can tell - it's actually initialized to `False` in the base `Celery.__init__` but gets changed to `True` later in the startup process via `Worker.on_before_init` and/or`Worker.on_init_blueprint` calling the `trace.setup_worker_optimizations()` function.